pkg/driver/trace_driver.go

```diff
@@ -236,177 +154,136 @@ func (d *Driver) invokeFunction(metadata *InvocationMetadata) {
 		atomic.AddInt64(metadata.SuccessCount, 1)
 	} else {
 		atomic.AddInt64(metadata.FailedCount, 1)
-		atomic.AddInt64(&metadata.FailedCountByMinute[metadata.MinuteIndex], 1)
 	}
 }
 
-func (d *Driver) functionsDriver(list *list.List, announceFunctionDone *sync.WaitGroup,
-	addInvocationsToGroup *sync.WaitGroup, readOpenWhiskMetadata *sync.Mutex, totalSuccessful *int64,
-	totalFailed *int64, totalIssued *int64, recordOutputChannel chan *mc.ExecutionRecord) {
+func getTotalInvocationCount(perMinuteCount []int, traceDuration int, granularity common.TraceGranularity) int {
+	if len(perMinuteCount) == 0 {
+		return 0
+	}
+
+	if granularity == common.SecondGranularity {
+		traceDuration *= 60
+	}
+
+	sum := 0
+	for i := 0; i < traceDuration; i++ {
+		sum += perMinuteCount[i]
+	}
+
+	return sum
+}
+
+func (d *Driver) functionsDriver(list *list.List, announceFunctionDone *sync.WaitGroup, addInvocationsToGroup *sync.WaitGroup, totalSuccessful *int64, totalFailed *int64, totalIssued *int64, recordOutputChannel chan *mc.ExecutionRecord) {
+	defer announceFunctionDone.Done()
 
 	function := list.Front().Value.(*common.Function)
-	numberOfInvocations := 0
-	for i := 0; i < len(function.InvocationStats.Invocations); i++ {
-		numberOfInvocations += function.InvocationStats.Invocations[i]
+	invocationCount := getTotalInvocationCount(function.Specification.PerMinuteCount, d.Configuration.TraceDuration, d.Configuration.TraceGranularity)
```

Comment:
This number should already be equal to the size of the IAT array by its construction. The only thing that you change is the interpretation of trace duration with second granularity (trace duration is always in minutes). This actually creates a problem with the front end: we parse only ExperimentDuration (+ WarmupDuration + 1) numbers from the trace without multiplying it by 60.

I think part of the confusion is the perMinuteCounts. With second granularity, it is actually per-second ones (we just append it after each granularity, so it is exactly the same as InvocationCounts). But in the getTotalInvocationCount, you interpret them only as per-minute.

Reply:
Replaced with `invocationCount := len(function.Specification.IAT)`

Reply:
I put `invocationCount := len(IAT)`. Seems that we were not following the convention in the tests on what we pass as input arguments. The `cmd/loader.go` operates as you describe, but tests we passing something else.